docs/DeveloperGuide.md

```diff
@@ -1,34 +1,315 @@
+
+**Problem**: When the user wants to review a module, he/she should be able to type ___multiple lines
+of input___, as reviews are generally long and detailed.
+
+**Solution**: A `while` loop is used to take in multiple lines of input until the input contains
+the `/end` symbol. Anything typed after this `/end` symbol will be erased.
+
+**When overwriting this review with a new review**: The old review will be printed for user
+reference. The user will also be alerted that this action will delete his/her old review via a log.
+The log is printed out as a warning.
+
+
+### 4.02 Delete a Review
+
+This feature allows user to delete a review from the `module`.
+
+A general explanation of how this feature works:
+
+Deleting a review would __reset__ the `Review` *String* attribute of a `module` to a default value
+of:  
+`"You have not reviewed this module yet."`
+
+### 4.03 Add New Task
+
+This feature allows the user to add tasks of type `task`, `assignment`, `midterm`, `final exam` to a
+task list.
+
+All task types (except `Task`, which contains arguments 1, 2 and 5) contain 5 arguments upon
+creation:
+
+1) `Module Code` <br> Example: `CS2113T`
+2) `Task Description` <br> Example: `Finish v2.0`
+3) `Date` <br> Example: `2021-03-21`
+4) `Time` <br> Example: `23:59`
+5) `Message` <br> Example: `Good job!`
+
+Given below is the sequence diagram for the feature:  
+![Diagram](diagrams/addNewTask.png)
+
+A general explanation of how this feature works:
+
+When the user calls this feature, the application will prompt them to input the arguments required
+for task creation. This new task will then be stored into an `ArrayList` for future reference. The
+user has to input a valid format for the __date__ and __time__ field (refer to section 3.3.1 of User
+Guide or the example above), and our application will convert it to a more readable format for the
+user. Using the example above,
+
+- `2021-03-21` will be outputted as `Mar 21 2021`
+- `23:59` will be outputted as `11:59 PM`
+
+The user input for `Message` (argument 5 above) will be printed out when the user deleted the task,
+signalling completion.
+
+### 4.04 Delete a Task
+
+This feature allows the user to delete a task of type `task`, `assignment`, `midterm`, `final exam`
+from a task list.
+
+A general explanation of how this feature works:
+
+When the user calls this feature, the application will prompt them to choose the __task
+type__ (`task`, `assignment`, `midterm`, `final exam`) of the task they want to delete. All existing
+tasks in the `ArrayList` for the task type they chose will then be printed out, and the user has to
+input the index of the task they want to delete. The task they select will then be deleted from
+the `ArrayList`.
+
+The `Message` that the user inputted upon creation of this particular task will then be printed out.
+
+### 4.05 Add Zoom Link
+
+Given below is the sequence diagram for the feature:  
+![Diagram](diagrams/addZoomLink.png)
```

Comment:
Good Sequence diagram. Perhaps it is alright to end the sequence diagram at the bottom with just lifelines? Maybe there is no need to end it with the object "boxes"

![image](https://user-images.githubusercontent.com/77471508/113231841-a9864780-92ce-11eb-9ea1-e582d0a8d1ce.png)